torch/distributed/pipeline/sync/_balance/profile.py

```diff
@@ -38,16 +38,21 @@ def layerwise_sandbox(module: nn.Sequential, device: torch.device,) -> Generator
 def detach(batch: Batch) -> None:
     """Detaches from autograd graph."""
     for i, x in enumerate(batch):
+        if not torch.is_tensor(x):
+            continue
         batch[i] = x.detach().requires_grad_(x.requires_grad)
 
 
+
```

Comment:
nit: Should we keep 2 newlines between functions? (i.e. delete this added newline)